editor/components/block-settings-menu/html-converter.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+import { rawHandler, getBlockContent } from '@wordpress/blocks';
+import { compose } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+export function HTMLConverter( { block, onReplace, small, canUserUseUnfilteredHTML, role } ) {
+	if ( ! block || block.name !== 'core/html' ) {
+		return null;
+	}
+
+	const label = __( 'Convert to blocks' );
+
+	const convertToBlocks = () => {
```

Comment:
I guess it could be refactored, indeed. I don't mind one-way or the other and did that to mimic what existed in UnknownConverter. I'll leave it as it is, we can always refactor both later if necessary.

Reply:
It's a leaf component inside the dropdown which is not rendered by default, so this shouldn't impact performance. In general, having a local function inside `render` method isn't expected because it causes unnecessary re-renders.

Reply:
Grzegorz, https://github.com/WordPress/gutenberg/pull/7885 follows up the work on this PR and addresses some ideas you mentioned in this thread (share common code, move to `withDispatch`).